### ISSUE TITLE
Warn when assigning to a path from `this`, aka `this.foo.bar = 42` (need to use `set`)

### ISSUE THREAD

**Issue by kevinpschaaf** [PolymerLabs/polylint#47]
Generally, any assignments to paths on an element should be done through `set`, to ensure proper notification in the system.

OK:

```js
this.foo = 42;
```

Should warn:

```js
this.foo.bar = 42;
```

The above should instead be:

```js
this.set('foo.bar', 42);
```

**Comment by neonsoftware** [comment]
Nice +1

**Comment by MeTaNoV** [comment]
Jup! +1
